packages/hoppscotch-common/src/components/environments/ImportExport.vue

```diff
@@ -158,251 +60,209 @@ const environmentJson = computed(() => {
     const teamEnvironments = props.teamEnvironments.map(
       (x) => x.environment as Environment
     )
-    return JSON.stringify(teamEnvironments, null, 2)
+    return teamEnvironments
   } else {
-    return JSON.stringify(myEnvironments.value, null, 2)
+    return myEnvironments.value
   }
 })
 
-const createEnvironmentGist = async () => {
-  if (!currentUser.value) {
-    toast.error(t("profile.no_permission").toString())
-
-    return
-  }
-
-  try {
-    const res = await axios.post(
-      "https://api.github.com/gists",
-      {
-        files: {
-          "hoppscotch-environments.json": {
-            content: environmentJson.value,
-          },
-        },
-      },
-      {
-        headers: {
-          Authorization: `token ${currentUser.value.accessToken}`,
-          Accept: "application/vnd.github.v3+json",
-        },
+const HoppEnvironmentsImport: ImporterOrExporter = {
+  metadata: {
+    id: "import.from_json",
+    name: "import.from_json",
+    icon: IconFolderPlus,
+    title: "import.from_json",
+    applicableTo: ["personal-workspace", "team-workspace"],
+    disabled: false,
+  },
+  component: FileSource({
+    acceptedFileTypes: "application/json",
+    caption: "import.hoppscotch_environment_description",
+    onImportFromFile: async (environments) => {
+      const res = await hoppEnvImporter(environments)()
+
+      if (E.isLeft(res)) {
+        failedImport()
+        return
+      } else {
+        importSuccessful(res.right)
       }
-    )
-
-    toast.success(t("export.gist_created").toString())
-
-    platform.analytics?.logEvent({
-      type: "HOPP_EXPORT_ENVIRONMENT",
-      platform: "rest",
-    })
 
-    window.open(res.data.html_url)
-  } catch (e) {
-    toast.error(t("error.something_went_wrong").toString())
-    console.error(e)
-  }
+      emit("hide-modal")
```

Comment:
Do we don't want to hide the modal on failure? Also, isn't the else redundant?

Reply:
When the import fails, the user might want to try again. so probably better to leave to leave the dialog open.